rancher/distros-test-framework
# Refactor dependency injection - cluster and env

#### Proposed Changes ####


- https://github.com/rancher/qa-tasks/issues/1069 




- This has many files changed cause it needed to change due to dependency, but whats changed in fact is very small changes:
- Env and Cluster dependencies

- Refactor how we inject dependencies in a way of create a more standardized and clear model which basically is:

    . Send dependencies top down injecting in the main and passing where needed

  Benefits:

   . Have a better visibility and understanding of your deps.
   . Have a easier way to follow through to debug and maintain
   . Standardize
   . Have deps available without need to duplicate code
   . Simplify process of adding dependencies ( now you just need to add and pass along in the main)


   


#### Types of Changes ####



#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

- yes 
2. Verify code lint; we should not have errors.

- yes 

3. Update the documentation if needed.

- will open a PR only for README 
4. Update makefile and docker run if adding new tests.
- n/a

5. Run your tests at least 4 times with all configurations needed and possible.
- yes 

6. If needed test with different os types.

- yes 
#### Linked Issues ####



#### Further Comments ####
